### ISSUE TITLE
php-cgi.exe crash working with stored procedure, php7 and sqlsrv_7

### ISSUE THREAD

**Issue by vovech** [Microsoft/msphpsql#186]
Hi People, i have similar issue using SQLServer with PDO, executing Store Procedure. Sometimes it happens with simple querys, not sure why. 

**PHP Versions:** 7.0.12.0 (i've tried with others)
**SQL Server:** 2008R2
**Extension:** extension=php_sqlsrv_7_ts_x86.dll
**IIS Version:** 8.5.9600.16384
**WIndows Server:** 2012R2

This is my store procedure:

`ALTER PROCEDURE [dbo].testSP 
AS
BEGIN
    SELECT 'SIMPLY TEST WITH SELECT' AS result
    return -1
END`

Here is my script:

```
error_reporting(E_ALL);
ini_set("display_errors", 1);
$pdo = new PDO("sqlsrv:Server={XXXX}; Database = {XXXX}", "XXXX", "XXXX", [PDO::ATTR_ERRMODE => PDO::ERRMODE_WARNING]);

$stmt = $pdo->prepare("EXEC testSP");
$stmt->execute();

do {
    $results[] = $stmt->fetchAll();
} while ($stmt->nextRowset());   

print_r($results);
```

_This works perfectly if you execute it in command line:_ 
**php-cgi.exe test.php**

_By using navigator:_
**www.mysite.com/test.php**

**IIS Server Throwing 500** error without any specific description, and event_log shows:

> Fault bucket , type 0
> Event Name: APPCRASH
> Response: Not available
> Cab Id: 0
> 
> Problem signature:
> P1: php-cgi.exe
> P2: 7.0.12.0
> P3: 57ffcd8f
> P4: php7ts.dll
> P5: 7.0.12.0
> P6: 57ffd3c2
> P7: c0000005
> P8: 00006e8c
> P9: 
> P10: 
> 
> Attached files:
> C:\Windows\Temp\WER1044.tmp.appcompat.txt
> C:\Windows\Temp\WER1064.tmp.WERInternalMetadata.xml
> C:\ProgramData\Microsoft\Windows\WER\ReportQueue\AppCrash_php-cgi.exe_1aec6c868dca22ed10abf8bc4a683b26443662f_476a3a53_cab_53201073\memory.hdmp
> C:\ProgramData\Microsoft\Windows\WER\ReportQueue\AppCrash_php-cgi.exe_1aec6c868dca22ed10abf8bc4a683b26443662f_476a3a53_cab_53201073\triagedump.dmp
> WERGenerationLog.txt

**Another Error:**

> Faulting application name: php-cgi.exe, version: 7.0.12.0, time stamp: 0x57ffcd8f
> Faulting module name: php7ts.dll, version: 7.0.12.0, time stamp: 0x57ffd3c2
> Exception code: 0xc0000005
> Fault offset: 0x00006e8c
> Faulting process id: 0xc48
> Faulting application start time: 0x01d2444406f7cf03
> Faulting application path: C:\Program Files (x86)\PHP\v5.7\php-cgi.exe
> Faulting module path: C:\Program Files (x86)\PHP\v5.7\php7ts.dll
> Report Id: 44b046b9-b037-11e6-80e0-00155d3bf438
> Faulting package full name: 
> Faulting package-relative application ID: 

I've tried to reproduce same error using XAMPP last versión with php7 and Windows 10. 

Here is a little bit different error. With script i've posted works and with a biggest Store Procedure it crash again: 

**ERR_CONNECTION_RESET** 

Windows log show:

> 
> Nombre de la aplicación con errores: httpd.exe, versión: 2.4.23.0, marca de tiempo: 0x577631b1
> Nombre del módulo con errores: php7ts.dll, versión: 7.0.13.0, marca de tiempo: 0x58224f14
> Código de excepción: 0xc0000005
> Desplazamiento de errores: 0x00006e9c
> Identificador del proceso con errores: 0xa94
> Hora de inicio de la aplicación con errores: 0x01d2445048af3f7e
> Ruta de acceso de la aplicación con errores: C:\xampp7\apache\bin\httpd.exe
> Ruta de acceso del módulo con errores: C:\xampp7\php\php7ts.dll
> Identificador del informe: 8380d68c-d983-4a04-9527-191215a2eea8
> Nombre completo del paquete con errores: 
> Identificador de aplicación relativa del paquete con errores: 
> 
>

**Comment by vovech** [comment]
Hi ulvii,

**php --ri pdo_sqlsrv**

> 
> pdo_sqlsrv
> 
> pdo_sqlsrv support => enabled
> ExtensionVer => 4.0.8629.3
> 
> Directive => Local Value => Master Value
> pdo_sqlsrv.log_severity => 0 => 0
> pdo_sqlsrv.client_buffer_max_kb_size => 10240 => 10240

**Comment by ulvii** [comment]
Could you also please make sure you are using PHP 7? 
From the log that you have posted above:  `Faulting application path: C:\Program Files (x86)\PHP\v5.7\php-cgi.exe`

**Comment by vovech** [comment]
Yes, i'm sure. This was an error to naming directory:
php -v
PHP 7.0.9 (cli) (built: Jul 20 2016 10:47:41) ( NTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies

**Comment by vovech** [comment]
**php-cgi.exe -v**

> 7.0.9 (cgi-fcgi) (built: Jul 20 2016 10:47:38)
> Copyright (c) 1997-2016 The PHP Group
> Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies

**Comment by vovech** [comment]
About second script, is the same PHP code, the difference is i'm using real Stored Procedure wich has more operations and CPU expensive.  I think that is imposible to share and make it work without entire database.

**Comment by vovech** [comment]
I also checked that IIS is pointing to right php versión and it's ok:

C:\Program Files (x86)\PHP\v5.7\php-cgi.exe
_(the name if directory is wrong, but is php7,  5.7 does not exists as versión)_

**Comment by vovech** [comment]
One more thing, there was a little bit error with my path than was pointing to another php7 versión. I fixed it and tested again with the same results.

**php -v**
```
PHP 7.0.12 (cli) (built: Oct 13 2016 11:04:23) ( ZTS )
Copyright (c) 1997-2016 The PHP Group
Zend Engine v3.0.0, Copyright (c) 1998-2016 Zend Technologies
```

**Comment by vovech** [comment]
Maby this helps, testing in cmd with different versions:

**./v5.6/php.exe test_full.php**

> Array
> (
>     [0] =>
>     [1] =>
>     [2] =>
>     [3] =>
>     [4] =>
>     [5] =>
>     [6] =>
>     [7] =>
>     [8] =>
>     [9] => Array
>         (
>             [txt_error] => TEST
>         )
>     [10] =>
> )

**./v7.0.12/php.exe test_full.php** Shows:

> Segmentation fault

line causing error: $stmt->fetch(); <-- Works in 5.6

**Comment by ulvii** [comment]
Hi @vovech 

We are unable to reproduce the issue by running the script provided on an environment with the following configurations:

```
PHP: 7.0.12 ZTS (NTS)
SQL Server: 2008R2
Extension: php_pdo_sqlsrv_7_ts.dll (php_pdo_sqlsrv_7_nts.dll)
IIS Version: 8.5.9600.16384
Windows Server: 2012R2
```

We would suggest to clean install PHP and check the IIS settings.

**Comment by vovech** [comment]
I just solved this problem with msphpsql 4.1.4-Windows (Pre-release), now works perfectly.

Thanks.
